sw/airborne/modules/sensors/electric_ardrone2.c

```diff
@@ -144,4 +146,10 @@ void electrical_periodic(void)
    */
   electrical.current = b - pow((pow(b, electrical_priv.nonlin_factor) - pow((b * x), electrical_priv.nonlin_factor)),
```

Comment:
This is already calculated in electrical_periodic, you should only need to update `electrical.vsupply` here...